core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
You assume you know the exact version of gRPC being used. As it is today, just by looking at the backtrace I could figure out what is wrong, and not even need to know the version being run. That applies even if the name becomes outdated. Most issues that are filed with a backtrace don't include the version of the library that experienced the problem. As far as I'm concerned, for internal APIs the string could be random and still serve much of its purpose. For public APIs, having the string name match the argument is very helpful to users as it means they don't need to read the source.

I read `Precoditions.checkNotNull` as a single token; it doesn't take me any longer to read it than just `checkNotNull` and I don't have to assume that the method was statically imported (which is more applicable for `checkState`). I used to do static imports for Preconditions but stopped, because 1) it was causing frequent import noise (both additions and removals), especially with checkState, and 2) that import noise was causing merge conflicts. The banning of static wildcard imports is essentially the cause of my prefixing. I'm not against doing the same for `org.junit.Assert`, except the JUnit 3 history and the redundant "assert" in the method name.